### PR TITLE
[Feature](Nereids)add rule: PruneOlapScanTablet

### LINKED ISSUE #14465: [Enhancement] (Nereids) bucket prune
### Search before asking

- [X] I had searched in the [issues](https://github.com/apache/doris/issues?q=is%3Aissue) and found no similar issues.

### Description

prune bucket use filter conjunts on OlapScanNode

### Solution

_No response_

### Are you willing to submit PR?

- [ ] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://www.apache.org/foundation/policies/conduct)

### PR DESCRIPTION
**Pull request by sohardforaname**
# Proposed changes

Issue Number: close #14465

## Problem summary

add rule: prune olap scan tablet

## Checklist(Required)

1. Does it affect the original behavior: 
    - [ ] Yes
    - [x] No
    - [ ] I don't know
2. Has unit tests been added:
    - [x] Yes
    - [ ] No
    - [ ] No Need
3. Has document been added or modified:
    - [ ] Yes
    - [ ] No
    - [x] No Need
4. Does it need to update dependencies:
    - [ ] Yes
    - [x] No
5. Are there any changes that cannot be rolled back:
    - [ ] Yes (If Yes, please explain WHY)
    - [x] No

## Further comments

If this is a relatively large or complex change, kick off the discussion at [dev@doris.apache.org](mailto:dev@doris.apache.org) by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

**Comment by hello-stephen** [comment]
TeamCity pipeline, clickbench performance test result:
 the sum of best hot time: 34.55 seconds
 load time: 441 seconds
 storage size: 17150900847 Bytes
 https://doris-community-test-1308700295.cos.ap-hongkong.myqcloud.com/tmp/20221120091918_clickbench_pr_49288.html

**Review by morrySnow — COMMENTED** [review]
please add:
1. regression test
2. UT for static function add Rule. NOTICE: UT should start from a **PLAN** and check the plan after transform. instead of start from a **SQL**

File: fe/fe-core/src/main/java/org/apache/doris/nereids/trees/expressions/visitor/ExpressionColumnFilterConverter.java [hunk 1]
```diff
@@ -0,0 +1,108 @@
...
+    @Override
+    public Expression visitComparisonPredicate(ComparisonPredicate predicate,
+            Map<String, PartitionColumnFilter> context) {
+        if (predicate instanceof NullSafeEqual) {
+            return null;
+        }
+        PartitionColumnFilter filter = new PartitionColumnFilter();
+        LiteralExpr literal = ((Literal) predicate.right()).toLegacyLiteral();
+        if (predicate instanceof EqualTo) {
+            setFilter(filter, literal, true, literal, true);
```
[the remark this answers is not in the document]
Comment (sohardforaname) [reply fe/fe-core/src/main/java/org/apache/doris/nereids/trees/expressions/visitor/ExpressionColumnFilterConverter.java:62]: no, the filter will be get from map and set again, so filter generator is not a good idea.

[the remark this answers is not in the document]
Comment (morrySnow) [reply fe/fe-core/src/main/java/org/apache/doris/nereids/trees/expressions/visitor/ExpressionColumnFilterConverter.java:63]:
```java
    private void updateFilter(String columnName, LiteralExpr lowerBound, boolean lowerInclusive,
            LiteralExpr upperBound, boolean upperInclusive) {
        PartitionColumnFilter filter = columnFilterMap.computeIfAbsent(columnName,
                k -> new PartitionColumnFilter());
        if (lowerBound != null) {
            filter.setLowerBound(lowerBound, lowerInclusive);
        }
        if (upperBound != null) {
            filter.setUpperBound(upperBound, upperInclusive);
        }
    }
```
and call it from other method

File: fe/fe-core/src/main/java/org/apache/doris/nereids/trees/expressions/visitor/ExpressionColumnFilterConverter.java [hunk 2]
```diff
@@ -0,0 +1,109 @@
...
+        } else if (predicate instanceof GreaterThan) {
+            setFilter(filter, literal, false, null, false);
+        } else if (predicate instanceof GreaterThanEqual) {
+            setFilter(filter, literal, true, null, false);
+        } else if (predicate instanceof LessThan) {
+            setFilter(filter, null, false, literal, false);
+        } else if (predicate instanceof LessThanEqual) {
+            setFilter(filter, null, false, literal, true);
```
Comment (morrySnow) [inline fe/fe-core/src/main/java/org/apache/doris/nereids/trees/expressions/visitor/ExpressionColumnFilterConverter.java:71]: u could use some local variables to record setFilter's parameters and call `setFilter` at the end once for easy reading

File: fe/fe-core/src/main/java/org/apache/doris/nereids/trees/expressions/visitor/ExpressionColumnFilterConverter.java (hunk 1 above)
[the remark this answers is not in the document]
Comment (sohardforaname) [reply fe/fe-core/src/main/java/org/apache/doris/nereids/trees/expressions/visitor/ExpressionColumnFilterConverter.java:63]: Yes!

**Review by morrySnow — APPROVED** [review]

**Pull request merged**